foundationdb/src/transaction.rs

```diff
@@ -979,16 +1014,19 @@ impl Future for TrxErrFuture {
 
 /// Futures for transaction, which supports retry/backoff with `Database::transact`.
 struct TrxFuture {
-    trx: Option<Transaction>,
+    // Order of fields should not be changed, because Rust drops field top-to-bottom, and future
+    // should be dropped before transaction.
```

Comment:
I think we'll need a custom Drop impl on this type. And then you can enforce the order of the dropped components.